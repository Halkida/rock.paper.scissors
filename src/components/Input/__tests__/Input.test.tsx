import { render, fireEvent } from '@testing-library/react';
import * as renderer from 'react-test-renderer';
import { Input } from '../';

const propsToTest = {
  isValid: [true, false],
  errorText: ['Some error text', ''],
  type: ['number', 'text', 'password']
};

describe('Input component testing', () => {
  test('Input props', () => {
    Object.entries(propsToTest).forEach(([prop, value]) => {
      if (Array.isArray(value)) {
        value.forEach((item) => {
          const component = renderer.create(
            <Input {...{[prop]: item}}>Input</Input>
          );
          const tree = component.toJSON();
          expect(tree).toMatchSnapshot();
        });
      } else {
        const component = renderer.create(
          <Input {...{[prop]: value}}>Input</Input>
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      }
    });
  });

  test('Input change event', () => {
    const onChange = jest.fn();
    const { getByPlaceholderText } = render(
      <Input onChange={onChange} placeholder='testing events' />
    );
    fireEvent.change(getByPlaceholderText('testing events'), {
      target: {value: 'test'}
    });

    expect(onChange).toHaveBeenCalledWith('test');
    expect(onChange).toHaveBeenCalledTimes(1);
  });
});