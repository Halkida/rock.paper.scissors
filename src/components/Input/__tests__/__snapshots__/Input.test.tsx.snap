// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component testing Input props 1`] = `
<input
  className="input"
  onChange={[Function]}
  type="text"
>
  Input
</input>
`;

exports[`Input component testing Input props 2`] = `
Array [
  <input
    className="input input_invalid"
    onChange={[Function]}
    type="text"
  >
    Input
  </input>,
  "",
]
`;

exports[`Input component testing Input props 3`] = `
<input
  className="input"
  onChange={[Function]}
  type="text"
>
  Input
</input>
`;

exports[`Input component testing Input props 4`] = `
<input
  className="input"
  onChange={[Function]}
  type="text"
>
  Input
</input>
`;

exports[`Input component testing Input props 5`] = `
<input
  className="input"
  onChange={[Function]}
  type="number"
>
  Input
</input>
`;

exports[`Input component testing Input props 6`] = `
<input
  className="input"
  onChange={[Function]}
  type="text"
>
  Input
</input>
`;

exports[`Input component testing Input props 7`] = `
<input
  className="input"
  onChange={[Function]}
  type="password"
>
  Input
</input>
`;
